packages/gatsby/src/schema/print.ts

```diff
@@ -57,23 +84,23 @@ const printTypeDefinitions = ({ config, schemaComposer }) => {
   const typesToExclude = exclude.types || []
   const pluginsToExclude = exclude.plugins || []
 
-  const getName = tc => tc.name || tc.getTypeName()
+  const getName = (tc: NamedTypeComposer<any>): string => tc.getTypeName()
 
-  const isInternalType = tc => {
+  const isInternalType = (tc: NamedTypeComposer<any>): boolean => {
```

Comment:
same here
```suggestion
  const isInternalType = (tc: NamedTypeComposer<unknown>): boolean => {
```

Reply:
Adjusted in [`727629d` (#34787)](https://github.com/gatsbyjs/gatsby/pull/34787/commits/727629d6b6b9167ce3d34e35c3a767034096ffc5)